### PR TITLE
Add an EmptyView In Bookmarks Activity

### LINKED ISSUE #1010: Add an EmptyView To Bookmarks Activity
**Is your feature request related to a problem? Please describe.**

It's a good practice to have an empty view. In bookmarks activity, there's absence of an empty view.

**Describe the solution you'd like**

* Adding an emptyview to the bookmarks activity.

**Additional context**

Currently, here's how it looks. 

For example:
Here's an example of a good empty view.

We can have an image that matches with our app's design colors. (i.e. B&W) and also one for the NIGHT MODE or we can have TEXT only. 
>**_I'm looking for suggestions here._**

### PR DESCRIPTION
**Pull request by iadeelzafar**
### Changes:
* Added a helper class RecyclerViewEmptySupport to attach the recyclerView with a AdapterDataObserver.
* Observer base class for watching changes to an RecyclerView.Adapter. 
[Reference](https://developer.android.com/reference/android/support/v7/widget/RecyclerView.AdapterDataObserver)
* Look for changes immediately after bookmarks are deleted.
* EmptyView Layout in the activity_bookmarks_history_language.xml
* String values for headings
* Vector image layout.

### CODE REVIEW

**Comment by iadeelzafar** [comment]
@abdulwd one thing that I need help with is this. I tried different solutions to change SVG image's color for night mode but I couldn't find this working.

**Comment by iadeelzafar** [comment]
Also, I wanted to know where should I give credits to the creator of this vector?

File: app/build.gradle [hunk 1]
```diff
@@ -30,7 +30,6 @@ repositories {
 String[] archs = ['arm64-v8a', 'armeabi', 'mips', 'mips64', 'x86', 'x86_64']
 
 dependencies {
-
```
Comment (mhutti1) [inline app/build.gradle]: Please don't commit files you havn't touched

Comment (mhutti1) [reply app/build.gradle]: Can you remove this and force push

**Comment by soloturn** [comment]
> Also, I wanted to know where should I give credits to the creator of this vector?
where did you get it from @iadeelzafar ?

**Comment by soloturn** [comment]
> Also, I wanted to know where should I give credits to the creator of this vector?

where did you get it from @iadeelzafar ?

**Comment by abdulwd** [comment]
I think you should choose a more visually appealing image and there should be consistency among all the images used in the empty states (eg for no history there will be another image).

**Comment by iadeelzafar** [comment]
@soloturn I got the image from [Flaticon](flaticon.com)

**Comment by iadeelzafar** [comment]
@abdulwd Alright. I'll search for a better image and could you please comment on the Night mode issue that I talked about above.

File: app/build.gradle (hunk 1 above)
Comment (iadeelzafar) [reply app/build.gradle]: @mhutti1 Thanks for reviewing. I'll do force push soon after making all the changes that are required.

**Comment by macgills** [comment]
@iadeelzafar can we close this? The solution is a bit complex for what amounts to setting the visiblity of a textview with compound drawable. Subclassing Recyclerview is rarely an avenue you want to go down.

**Comment by iadeelzafar** [comment]
Apologies for not being active on this ticket. @macgills if you or anyone wants to pick this up, go for it. Or if you think working on it's solution is not worth it, feel free to close it.

**Pull request closed without merging**